### PR TITLE
Improvements and some simple refactorings

### PR DESCRIPTION
**Pull request by luizfernandonb**
Any mistake or doubt, just talk to me

### CODE REVIEW

File: SharpSpades/Utils/Throw.cs [hunk 1]
```diff
@@ -6,22 +6,16 @@ namespace SharpSpades.Utils
...
     internal static class Throw
     {
-        public static void IfNull<T>(T? obj) where T : class
+        public static void If<T>(T? obj, Predicate<T?> predicate, Exception exception) //where T : class
         {
-            if (obj is null)
-                throw new ArgumentNullException();
+            if (predicate.Invoke(obj))
+                throw exception;
         }
```
Comment (JStalnac) [inline SharpSpades/Utils/Throw.cs]: I think it's a bit redundant having a method that throws a given exception if a given condition isn't met as it's what regular if-statements do and the code doesn't really become much cleaner with it.

Also having to initialize a new exception object every time you want to use the helper method is propably not very good.

File: SharpSpades/Net/Client.cs [hunk 2]
```diff
@@ -236,17 +246,14 @@ public async ValueTask SendPacket(Packet packet)
         /// Disconnects the client.
         /// </summary>
         /// <param name="reason"></param>
-        public ValueTask DisconnectAsync(DisconnectReason reason)
+        public ValueTask Disconnect(DisconnectReason reason)
```
Comment (JStalnac) [inline SharpSpades/Net/Client.cs]: This should be `DisconnectAsync` as the result is awaitable.

**Review by JStalnac — CHANGES_REQUESTED** [review]
Looking great, good point add the `Async` suffix to `SendMap` and `SendPlayers`.

General note: code that checks if a parameter is null should throw an `ArgumentNullException`, not a `NullReferenceException`

**Comment by JStalnac** [comment]
Our editors/IDEs seem to have different code style configurations. I'll push a .editorconfig file tomorrow to address that problem.

**Comment by luizfernandonb** [comment]
> Our editors/IDEs seem to have different code style configurations. I'll push a .editorconfig file tomorrow to address that problem.

Ok

**Review by JStalnac — CHANGES_REQUESTED** [review]
Good, one last thing: could you add back the `IfNull(value, paramName)` and `IfNull(value)` helper methods that throw the `ArgumentNullException` with the default message and maybe have e. g. packets use those instead of having a custom message?

Thank you for your patience!

**Comment by JStalnac** [comment]
I'll do the code style thing on my own, you don't have to worry about it. I've been busy so I haven't had time to set up the editorconfig.

**Comment by luizfernandonb** [comment]
> Good, one last thing: could you add back the `IfNull(value, paramName)` and `IfNull(value)` helper methods that throw the `ArgumentNullException` with the default message and maybe have e. g. packets use those instead of having a custom message?
> 
> Thank you for your patience!

I didn't understand this: `and maybe have e. g. packets use those instead of having a custom message?`
Can you explain again?

**Comment by luizfernandonb** [comment]
> > Good, one last thing: could you add back the `IfNull(value, paramName)` and `IfNull(value)` helper methods that throw the `ArgumentNullException` with the default message and maybe have e. g. packets use those instead of having a custom message?
> > Thank you for your patience!
> 
> I didn't understand this: `and maybe have e. g. packets use those instead of having a custom message?`
> Can you explain again?

As I understand it, are you talking about using a NuGet package to launch a personalized message when an exception is thrown? If so, unfortunately I don't have a package

**Comment by JStalnac** [comment]
Sorry if I was a bit unclear. I'm talking about using the `IfNull(value, paramName)` method in place of `IfNull(value, paramName, message)` which takes a custom message from `StringUtils.GenerateNullExceptionMessage` because the default message for `ArgumentNullException` works fine and results in less code.

**Comment by JStalnac** [comment]
> Sorry if I was a bit unclear. I'm talking about using the `IfNull(value, paramName)` method in place of `IfNull(value, paramName, message)` which takes a custom message from `StringUtils.GenerateNullExceptionMessage` because the default message for `ArgumentNullException` works fine and results in less code.

I don't know how to explain this shortly :)

**Review by JStalnac — APPROVED** [review]
Great! Looking good! Thank you for your time and patience!

**Pull request merged**